docs-search/common/src/Docs/Search/Declarations.purs

```diff
@@ -110,80 +121,82 @@ resultsForDeclaration scores moduleName indexEntry@(Declaration entry) =
 
 mkInfo :: DeclLevel -> Declaration -> Maybe ResultInfo
 mkInfo declLevel (Declaration { info, title }) =
-  case info.declType of
-
-    DeclValue ->
-      info."type" <#> \ty -> ValueResult { type: ty }
-
-    DeclData ->
-      make <$> info.typeArguments <*> info.dataDeclType
-      where
-      make typeArguments dataDeclType =
-        DataResult { typeArguments, dataDeclType }
-
-    DeclExternData ->
-      info.kind <#> \kind -> ExternDataResult { kind }
-
-    DeclTypeSynonym ->
-      make <$> info."type" <*> info.arguments
-      where
-      make ty args = TypeSynonymResult { "type": ty, arguments: args }
-
-    DeclTypeClass
-      | Just fundeps <- info.fundeps
-      , Just arguments <- info.arguments
-      , Just superclasses <- info.superclasses ->
-          Just $ TypeClassResult { fundeps, arguments, superclasses }
-      | otherwise -> Nothing
-
-    DeclAlias ->
+  case info of
+    ValueDeclaration ty ->
+      Just $ ValueResult { type: ty }
+
+    DataDeclaration dataDeclType typeArguments _ ->
+      Just $ DataResult
+        { dataDeclType
+        , typeArguments: typeArguments <#> toTypeArgument
+        }
+
+    ExternDataDeclaration kind _ ->
+      Just $ ExternDataResult { kind }
+
+    TypeSynonymDeclaration arguments ty ->
+      Just $ TypeSynonymResult
+        { type: ty
+        , arguments: arguments <#> toTypeArgument
+        }
+
+    TypeClassDeclaration arguments superclasses fundeps ->
+      Just $ TypeClassResult
+        { fundeps
+        , arguments: arguments <#> toTypeArgument
+        , superclasses
+        }
+
+    AliasDeclaration _ _ ->
       case declLevel of
         TypeLevel -> Just TypeAliasResult
         ValueLevel -> Just ValueAliasResult
-        _ -> Nothing
 
-    DeclExternKind ->
-      Just ExternKindResult
+  where
+  toTypeArgument :: Tuple _ _ -> TypeArgument
+  toTypeArgument (Tuple name kind) = { name, kind }
 
 -- | Level of a declaration, used to determine which URI hash anchor to use in
--- | links ("v", "t" or "k").
-data DeclLevel = ValueLevel | TypeLevel | KindLevel
+-- | links ("v" or "t" ).
+data DeclLevel = ValueLevel | TypeLevel
 
 declLevelToHashAnchor :: DeclLevel -> String
 declLevelToHashAnchor = case _ of
   ValueLevel -> "v"
   TypeLevel -> "t"
-  KindLevel -> "k"
```

Comment:
Distinguishing between types and kinds is not necessary since they share a namespace. 
We only need hash anchors to distinguish types and values. 

Also because the syntax for declaring kinds has changed from `foreign import kind Foo` 
to `foreign import data Foo :: Type` it would introduce unneeded complexity in the code to distinguish the type level from the kind level.